src/satosa/backends/saml2.py

```diff
@@ -149,6 +149,16 @@ def authn_request(self, context, entity_id):
         :param entity_id: Target IDP entity id
         :return: response to the user agent
         """
+
+        # If IDP blacklisting is enabled and the selected IDP is blacklisted,
+        # stop here
+        if self.config["sp_config"].get("idp_blacklist_enabled", None):
```

Comment:
The place where the setting is set, is in `pysaml2`'s configuration domain. As such it should be registered with `pysaml2`'s configuration module, and accessed as `self.sp.config.getattr("idp_blacklist_enabled", "")`. 

If we intend this to be a SATOSA configuration option (which makes sense because it enforces policy) it should be moved out of the `sp_config` block, and placed under the `config` block. Then, one can either set it on initialization (through `__init__` and use `self. idp_blacklist_enabled`) or invoke the configuration directly as `config.get("idp_blacklist_enabled")`.